### PR TITLE
[Process] Fix broken tests for PHP 7.2

### PR DESCRIPTION
**Pull request by sroze**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no 
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | no
| Fixed tickets | #24524, #24515
| License       | MIT
| Doc PR        | ø

Following #24515, trying to fix Process tests with PHP 7.2

### CODE REVIEW

**Comment by sroze** [comment]
Status: needs work
Reason: tests are not actually running and it's only running PHP 7.2 for now

File: .travis.yml [hunk 1]
```diff
@@ -20,11 +20,7 @@ env:
 
 matrix:
     include:
-        - php: 7.1.3
-        - php: 7.1
-          env: deps=high
-        - php: 7.1
-          env: deps=low
+        - php: 7.2
```
Comment (nicolas-grekas) [inline .travis.yml]: can you add another line: 
```
- php: 7.2
- php: 7.2
  env: deps=low
```

**Comment by stof** [comment]
you should avoid it being marked as an intermediary version though. You don't want it to be skipped.

**Comment by sroze** [comment]
@stof do you think we should prevent 7.2 to be marked as intermediate for all the PRs or it was just in the context of my debugging?

**Comment by sroze** [comment]
Status: Needs Review

File: src/Symfony/Component/HttpFoundation/Session/Storage/NativeSessionStorage.php [hunk 2]
```diff
@@ -88,17 +88,24 @@ class NativeSessionStorage implements SessionStorageInterface
...
-        $options += array(
-            // disable by default because it's managed by HeaderBag (if used)
-            'cache_limiter' => '',
-            'use_cookies' => 1,
-        );
+        $this->setMetadataBag($metaBag);
+
+        if (\PHP_SESSION_ACTIVE === session_status()) {
+            if (!empty($options) || null !== $handler) {
+                throw new \LogicException('Cannot change options or handler of an active session');
```
Comment (Simperfit) [inline src/Symfony/Component/HttpFoundation/Session/Storage/NativeSessionStorage.php]: Do we really to throw here ? maybe a test should be added to test this behaviour ?

File: .travis.yml [hunk 3]
```diff
@@ -16,15 +16,16 @@ addons:
...
-        - SYMFONY_PROCESS_PHP_TEST_BINARY=~/.phpenv/versions/7.1/bin/php
+        - SYMFONY_PROCESS_PHP_TEST_BINARY=~/.phpenv/shims/php
 
 matrix:
     include:
         - php: 7.1.3
         - php: 7.1
           env: deps=high
-        - php: 7.1
+        - php: 7.2
```
Comment (Simperfit) [inline .travis.yml]: maybe we could add without deps=low too

Comment (nicolas-grekas) [reply .travis.yml]: nope: the matrix is optimized to test the maximum of cases with the minimum number of jobs

**Comment by sroze** [comment]
@nicolas-grekas as discussed, kept here only the process test change and travis configuration and moved the native session storage PR to https://github.com/symfony/symfony/pull/24531 to target 2.7.

**Comment by sroze** [comment]
And rebased.

**Comment by sroze** [comment]
Rebased and now 💚. 

/cc @xabbuh

**Review by xabbuh — APPROVED** [review]

**Review by chalasr — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @sroze.

**Pull request closed without merging**